### DISCUSSION TITLE
特定のキーが反応しません@keyball39

### DISCUSSION THREAD

**Discussion by bakunyo** [Yowkees/keyball#193] [Q&A]
Keyball39 、右手トラックボールの際の左手側の以下のキーが反応しません。

<img width="860" alt="スクリーンショット 2023-01-27 21 36 13" src="https://user-images.githubusercontent.com/2487437/215997266-d31930dd-c268-4d95-b476-f9236b5897a6.png">

親指はロープロファイルを使用しています。
キーソケットのはんだ付け不良かと思い付け足したりしたのですが、状況変わらずです。
![keyball39_noballside](https://user-images.githubusercontent.com/2487437/215998583-707ab66d-6a06-4eaf-a50a-f7e266bd3dc8.jpeg)

ダイオードの向きは確認し直しました。
また、 https://github.com/Yowkees/keyball/discussions/57 のコメントを参照し、COL5とROW3にテスターを当てて確認したところ導通確認はできました。

他に確認すべきところ、対処方法などはありますでしょうか？

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

導通が確認できたということは、キーソケットの両端子をピンセット等でショートさせた場合は入力されるということでしょうか？

入力できるようでしたら、キーソケットかスイッチの不具合の可能性が濃厚ですので部品の交換をお願いします。

入力できない場合には、ProMicro裏の半田付けを見直してください。
12Pピンソケットのはんだ付けは、スルーホールがジグザグに配置されていて、隣との距離が近いのでショートしやすいです。
ここも確認してみてください。

以上、お手数ですがご確認お願いいたします。

**Reply by bakunyo** [reply]
コメントありがとうございます。

> キーソケットの両端子をピンセット等でショートさせた場合は入力されるということでしょうか？

こちら試したところ、入力されませんでした。

> 入力できない場合には、ProMicro裏の半田付けを見直してください。
12Pピンソケットのはんだ付けは、スルーホールがジグザグに配置されていて、隣との距離が近いのでショートしやすいです。

確かにジグザグの隣と近い場所があったので、はんだ吸い取り線を使って距離を確保するようにいたしました。
しかし、再度 Test Matrix を使って確認しても状況は変わらずでした。

（正しいかわかりませんが）別の確認事項として、キーソケットの両端にテスターの先をつけて、キーを押したときは導通することを確認しました。

素人のためどこをどう確認すればよいかわからず恐縮ですが、他に確認できそうな箇所があれば教えて頂ければ幸いです。

**Comment by Yowkees** [comment]
・キーソケットの両端をピンセットでショートしても入力できない
・キーソケットの両端にテスターの先をつけてキーを押したときは導通する
以上の２点が正しければ、やはり、キーソケットのテンプラ（キーソケット端子と基板のパッドが導通していない）が原因かと思います。
キーソケットの端子を横から覗いてみて、端子とパッドの間に半田が充填されているかご確認ください。
判断が難しい場合はお写真を見せていただければアドバイスできるかも知れません。

以上、お手数ですがご対応お願いいたします。

**Reply by bakunyo** [reply]
ありがとうございます。
見た目としては半田が充填されているように見えたのですが、あらためてキーソケットを付け直してみました。
写真を添付します（汚くてすみません 🙇 ）
<img width="50%" src="https://user-images.githubusercontent.com/2487437/216741230-f7962019-cee6-4eef-99cf-78f0854fbea9.jpeg" /><img width="50%" src="https://user-images.githubusercontent.com/2487437/216741238-2dfbf432-d3db-4953-9f94-21ce3eee1b47.jpeg" />

ただ、付け直したのが原因なのか、以下の確認が取れなくなってしまいました 💦 

> ・キーソケットの両端にテスターの先をつけてキーを押したときは導通する

また、（これは元からなのですが）キーソケットの両端子をピンセットでショートして入力されるか確認する際に、CherryMX互換ソケットの両端をショートしても入力されないというのも気になっています（他の親指キーは入力される）。

**Comment by bakunyo** [comment]
こちら、無事解決しすべてのキーが反応するようになりました 🙇‍♂️ 
原因は、 ProMicro の 12（COL5）のピンとピンソケットの対応部分が導通していなかったことでした 💦 

ProMicro を改めてしっかり差し込む（ゆるく差してるつもりはなかったですが）ことで COL5 部分が導通し、キーが反応するようになりました。
ご助言いただき、ありがとうございました！
